config/config.go

```diff
@@ -432,33 +455,53 @@ func (s *SourceBlockchain) Validate(destinationBlockchainIDs *set.Set[string]) e
 	}
 	s.subnetID = subnetID
 
-	// Validate and store the allowed destinations for future use
-	s.supportedDestinations = set.Set[ids.ID]{}
-
 	// If the list of supported destinations is empty, populate with all of the configured destinations
 	if len(s.SupportedDestinations) == 0 {
 		for _, blockchainIDStr := range destinationBlockchainIDs.List() {
-			blockchainID, err := ids.FromString(blockchainIDStr)
-			if err != nil {
-				return fmt.Errorf("invalid blockchainID in configuration. error: %w", err)
-			}
-			s.supportedDestinations.Add(blockchainID)
+			s.SupportedDestinations = append(s.SupportedDestinations, &SupportedDestination{
+				BlockchainID: blockchainIDStr,
+			})
 		}
 	}
-
-	for _, blockchainIDStr := range s.SupportedDestinations {
-		blockchainID, err := ids.FromString(blockchainIDStr)
+	zeroAddress := common.Address{}
```

Comment:
We unfortunately can't use `database.AllAllowedAddresses` here without introducing an import cycle. I don't want to move that variable to an upstream package, since it logically belongs to the `database` package. So I elected to explicitly disallow the zero address in the config, and not reference that variable here at all.